### PR TITLE
frontend/fix/ Customer feedback pills

### PR DESCRIPTION
**Pull request by aryan-13**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added feedback chips in rating card 
Android Output - [Link](https://drive.google.com/file/d/12pqzbCFRyvBQNxDBj2L8k4suQp2qlB5o/view?usp=sharing)
IOS Output - [Link](https://drive.google.com/file/d/1k-p5Zr1OAMfmDta_ZO2Yzz_R83sBND2p/view?usp=sharing)
Figma - [Link](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=61198-19125&mode=design&t=FQZD73HQDbUeJOrx-4)

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

**Review by witcher-shailesh — CHANGES_REQUESTED** [review]

**Review by ashishsingh101 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 1]
```diff
@@ -885,7 +888,16 @@ eval (OnIconClick autoAssign) state = do
...
-eval (RatingCardAC (RatingCard.Rating index)) state = continue state { data { rideRatingState { rating = index }, ratingViewState { selectedRating = index} } }
+eval (RatingCardAC (RatingCard.Rating index)) state = do 
+  let temp = if index == state.data.rideRatingState.rating then state.data.rideRatingState.feedbackList else []
+  _ <- pure $ spy "feedback lsit "  temp
+  continue state { data { rideRatingState { rating = index , feedbackList = if index == state.data.rideRatingState.rating then state.data.rideRatingState.feedbackList else []}, ratingViewState { selectedRating = index} } }
+
+eval (RatingCardAC (RatingCard.SelectPill feedbackItem id)) state = do
+  let newFeedbackList = updateFeedback id feedbackItem state.data.rideRatingState.feedbackList
+      filterFeedbackList = filter (\fa -> length fa.answer > 0) newFeedbackList
+  _ <- pure $ spy "feedback lsit "  filterFeedbackList
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:899]: @aryan-13 can you please remove spy statements.

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 2]
```diff
@@ -885,7 +888,16 @@ eval (OnIconClick autoAssign) state = do
 eval PreferencesDropDown state = do
   continue state { data { showPreferences = not state.data.showPreferences}}
 
-eval (RatingCardAC (RatingCard.Rating index)) state = continue state { data { rideRatingState { rating = index }, ratingViewState { selectedRating = index} } }
+eval (RatingCardAC (RatingCard.Rating index)) state = do 
+  let temp = if index == state.data.rideRatingState.rating then state.data.rideRatingState.feedbackList else []
+  _ <- pure $ spy "feedback lsit "  temp
+  continue state { data { rideRatingState { rating = index , feedbackList = if index == state.data.rideRatingState.rating then state.data.rideRatingState.feedbackList else []}, ratingViewState { selectedRating = index} } }
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:894]: @aryan-13 temp is defined already. Can we consume this variable.

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 3]
```diff
@@ -885,7 +888,16 @@ eval (OnIconClick autoAssign) state = do
...
+
+eval (RatingCardAC (RatingCard.SelectPill feedbackItem id)) state = do
+  let newFeedbackList = updateFeedback id feedbackItem state.data.rideRatingState.feedbackList
+      filterFeedbackList = filter (\fa -> length fa.answer > 0) newFeedbackList
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:898]: @aryan-13 please rename fa to something more meaningful.

**Review by MercyQueen — CHANGES_REQUESTED** [review]

**Review by ashishsingh101 — APPROVED** [review]

**Comment by kavya-shree-s** [comment]
@aryan-13 Is this tested in all the other language and other screen size device ?

**Review by Rohit4997 — APPROVED** [review]

**Pull request merged**